packages/SwingSet/src/kernel/vatAdmin/vatAdminWrapper.js

```diff
@@ -154,6 +154,9 @@ export function buildRootObject(vatPowers) {
 
   // the kernel sends this when the vat halts
   function vatTerminated(vatID, shouldReject, info) {
+    if (pending.has(vatID)) {
+      newVatCallback(vatID, { error: info });
```

Comment:
let's add a comment:

This happens when a dynamic vat survives `vatWarehouse.createDynamicVat()` but fails during the delivery of `startVat`. In that case, a successful `newVatCallback` was unwound by the failed delivery, but then it queues a `vatTerminated` for us. If this arrives before we see `newVatCallback`, that means the vat failed, and we should simulate the errorful `newVatCallback` so the caller will be notified about their misfortune.